domain/schema/model/sql/0017-machine.sql

```diff
@@ -27,6 +27,18 @@ ON machine (name);
 CREATE UNIQUE INDEX idx_machine_net_node
 ON machine (net_node_uuid);
 
+CREATE TABLE machine_parent (
+    machine_uuid TEXT NOT NULL PRIMARY KEY,
+    parent_uuid TEXT NOT NULL,
+    CONSTRAINT fk_machine_parent_machine
+    FOREIGN KEY (machine_uuid)
+    REFERENCES machine (uuid),
+    CONSTRAINT fk_machine_parent_parent
+    FOREIGN KEY (parent_uuid)
+    REFERENCES machine (uuid),
+    UNIQUE (parent_uuid)
+);
```

Comment:
`PRIMARY KEY (machine_uuid, parent_uuid)` ensures that you must only have one machine_uuid and primary_uuid per table. Attempting to add 1 (machine) and 2 (parent) is fine, adding 1 and 2 again, will be a unique primary constraint failure. This is the canonical example we've used elsewhere.

I think it's much cleaner in the prior change.

Reply:
Yes, but it doesn't prevent the following case:

* 1 (machine), 2 (parent)
* 1 (machine), 3 (parent)

meaning a machine with several parent, which is wrong, no ?

For me, 
* a machine can only have one parent, which can be represented as a PRIMARY KEY constraints (since we will fetch the parent through its child, using the primary index)
* if a parent can only have one child, we should also have a UNIQUE constraint on parent. 
* if one day we want to fetch quickly the child's of a machine we may need an index on parent either.

So for me, `PRIMARY KEY (machine_uuid, parent_uuid)` is wrong, even if it is equivalent of my first proposition, and there is a question over the unique constraint on `parent_uuid`, which may or may not be correct.

Reply:
> meaning a machine with several parent, which is wrong, no ?

True, can you add a comment as to why this is different?